tools/api-markdown-documenter/src/LintApiModel.ts

```diff
@@ -0,0 +1,140 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	ApiDocumentedItem,
+	type ApiItem,
+	ApiItemContainerMixin,
+	type ApiModel,
+} from "@microsoft/api-extractor-model";
+import type { DocInheritDocTag } from "@microsoft/tsdoc";
+
+import { defaultLoadModelOptions, loadModel, type LoadModelOptions } from "./LoadModel.js";
+import { noopLogger } from "./Logging.js";
+import { DocumentWriter } from "./renderers/index.js";
+import { resolveSymbolicReference } from "./utilities/index.js";
+
+/**
+ * Linter check options.
+ *
+ * @public
+ */
+export interface LinterOptions {
+	/**
+	 * Whether or not to evaluate `{@link}` and `{@inheritDoc}` links as a part of the linting process.
+	 * @defaultValue `true`
+	 */
+	checkLinks?: boolean;
+}
+
+/**
+ * {@link lintApiModel} options.
+ */
+export interface LintApiModelOptions extends LoadModelOptions, LinterOptions {}
+
+/**
+ * {@link LintApiModelOptions} defaults.
+ */
+const defaultLintApiModelOptions: Required<Omit<LintApiModelOptions, "modelDirectoryPath">> = {
+	...defaultLoadModelOptions,
+	checkLinks: true,
+};
+
+/**
+ * Validates the given API model.
+ *
+ * @throws If the specified {@link LoadModelOptions.modelDirectoryPath} doesn't exist, or if no `.api.json` files are found directly under it.
+ */
+export async function lintApiModel(options: LintApiModelOptions): Promise<void> {
```

Comment:
For reviewers: this is the API I plan to expose, and this file contains most of the related logic. Note that it isn't package-exported yet - that will come in a later PR. For now this just establishes the API internally and adds unit tests for it.